### ISSUE TITLE
Pages can get stuck updating

### ISSUE THREAD

**Issue by jkmassel** [wordpress-mobile/WordPress-iOS#17343]
Reported via ZD-4389760 – a page was saved on an unreliable internet connection and got stuck in the `.pushing` state without a clear way to resolve the issue. 

Workaround: Logging out and back in should fix the issue

**Comment by designsimply** [comment]
To try to help move this issue forward, what I would like to do is to list out the steps as given by the user to try to understand the problem better. Here is how they described the problem and the parts in square brackets were paraphrased by me to be shorter.

1. [WiFi goes offline.]
1. I edited and updated a page.
1. [WiFi comes back online.]
1. [Tried to access the My Site page in the WP app AKA dashboard.]
1. I kept getting an error message saying that the newspaper is currently being uploaded.
1. I did the same thing after 5 minutes, then the same video after 10 minutes.
1. I went into the app and tried to open the page, but still a lot of registrations, you can't right now edit the page is uploading, try again in a few minutes.

Note: the reporting user mentioned they are blind so you may need to test using VoiceOver even though that wasn't explicitly noted. 

<sup>(internal reference: 4389760-zd-woothemes)</sup>

Based on their description, it sounds to me like once a failed media upload happens, then it's impossible to get back into the app via VoiceOver. This isn't reflected in the steps they gave but in order to do it, I think you'd just need to trigger a media upload failure by turning off the network connection while media is uploading just prior to step 1. I don't think this would be limited to Pages fwiw.

**Comment by twstokes** [comment]
Possibly related, but I found that if I:

1. Draft a new page
2. Disable network traffic on the device (I'm using [Network Link Conditioner](https://nshipster.com/network-link-conditioner/) for the Simulator)
3. Publish the page (observe it only exists in Drafts)
4. Kill the app
5. Enable network traffic
6. Start the app
7. Tap on the page in Drafts and observe "This page is currently uploading. It won't take long - try again soon and you'll be able to edit it."

- The page may also exist under "Published" and can be edited without the "This page is currently uploading..." message
- Pulling to refresh and sync doesn't resolve the issue

This flow might be slightly different, though, because it read to me like the user was editing an existing page instead of creating a new one.

**Comment by kean** [comment]
Outdated – address in 24.9.
